src/screens/Wallet.js

```diff
@@ -99,19 +96,10 @@ function Wallet() {
     initializeWalletScreen();
   }, [selectedToken]);
 
-  // When the tokens history changes, check the last timestamp to define if we should fetch the token details again
+  // When the tokens history changes, update the token info
   useEffect(() => {
-    const selectedTokenHistory = get(tokensHistory, selectedToken, {
-      status: TOKEN_DOWNLOAD_STATUS.LOADING,
-      updatedAt: -1,
-      data: [],
-    });
-
-    if (tokenHistoryTimestamp !== selectedTokenHistory.updatedAt) {
-      setTokenHistoryTimestamp(selectedTokenHistory.updatedAt);
-      updateTokenInfo(selectedToken);
-      updateWalletInfo(selectedToken);
-    }
+    updateTokenInfo(selectedToken);
+    updateWalletInfo(selectedToken);
   }, [tokensHistory]);
```

Comment:
Is it possible to have here the `tokensHistory` of the `selectedToken`? Otherwise we would be updating token info and wallet info every time any token has its history updated.

Reply:
Yes, this is the downside of this approach.
I can test with `[selectedToken, tokensHistory[selectedToken]]` but react is not great at detecting changes on arrays/objects, we would need to check something specific like `[selectedToken, tokensHistory[selectedToken].updatedAt]` but this would re-enable the issue this is meant to fix.